src/praisonai-agents/tests/unit/test_gateway_rate_limit_policy.py

```
@@ -0,0 +1,106 @@
+    assert policy.check(identity="a", scope="auth", now=0.0).allowed is True
+    # Different identity, same scope: independent budget.
+    assert policy.check(identity="b", scope="auth", now=0.0).allowed is True
+    # Same identity, different scope: independent budget.
+    assert policy.check(identity="a", scope="approve", now=0.0).allowed is True
+    # Same identity + scope again: now limited.
+    assert policy.check(identity="a", scope="auth", now=1.0).allowed is False
+
+
+def test_decision_is_frozen():
+    decision = RateLimitDecision(allowed=False, retry_after_seconds=1.5)
+    with pytest.raises(Exception):
```

Ruff (0.15.20)

[warning] Do not assert blind exception: `Exception`

(B017)